plutus-tx/src/PlutusTx/Builtins/Class.hs

```diff
@@ -113,14 +107,11 @@ instance IsString BuiltinString where
     -- the application of 'stringToBuiltinString'
     {-# INLINE fromString #-}
     -- See Note [noinline hack]
-    fromString = Magic.noinline stringToBuiltinString
+    fromString str = Magic.noinline $ stringToBuiltinString (pack str)
 
 {-# INLINABLE stringToBuiltinString #-}
-stringToBuiltinString :: String -> BuiltinString
-stringToBuiltinString = go
-    where
-        go []     = emptyString
-        go (x:xs) = charToString x `appendString` go xs
+stringToBuiltinString :: Text -> BuiltinString
```

Comment:
delete, we should just use `toBuiltin` instead.

Reply:
Oh, we use it in the compiler, irritating. I guess leave it, I'll think about that later.